packages/connector-jsdom/src/connector.ts

```diff
@@ -293,9 +254,11 @@ export default class JSDOMConnector implements IConnector {
 
                     debug(`${this.finalHref} loaded, traversing`);
                     try {
-                        await this.server.emitAsync('traverse::start', event);
-                        await this.traverseAndNotify(window.document.children[0] as HTMLElement);
-                        await this.server.emitAsync('traverse::end', event);
+                        const html = await this.html;
+
+                        const asyncWindow = createAsyncWindow(html);
+
+                        await traverse((asyncWindow as JSDOMAsyncWindow).dom!, this.server, this.finalHref);
```

Comment:
Do we need the `JSDOMAsyncWindow` and such with the new approach? the cdp connector doesn't seem to need this.

Reply:
Working on it.